src/modules/messageHandler.js

```diff
@@ -27,19 +119,23 @@ function setupMessageHandlers() {
     const task = () => new Promise((resolve) => {
       if (request.action === "getQuantStats") {
         const symphony = request?.symphony;
+        const benchmarkData = request?.benchmarkData || null;
         log("Getting QuantStats");
         log("sym", symphony);
         log("dc", symphony?.dailyChanges);
 
-        const cacheKey = `quantstats_${symphony.id}`;
+        // Include benchmark hash in cache key if benchmarks provided
+        // v2: Using scipy linregress instead of qs.stats.greeks
+        const benchmarkSuffix = benchmarkData ? '_withBenchmarks_v2' : '';
```

Comment:
We should not overload this message. instead we should create a message key and have it handle all this cases special stuff.